### PR TITLE
feat: adds headers for choose and filter

### LINKED ISSUE #278: [Feature request] Add `--header` for `choose` and `filter`
just like
- #80 
- #216

### PR DESCRIPTION
**Pull request by MikaelFangel**
Fixes #278

### Changes
- Adds headers for both filter and choose where I've used #253 as a reference.

I'm not necessarily happy with the flow of the filter View functions return, as it's a bit cluttered.

### CODE REVIEW

**Comment by maaslalani** [comment]
I'm not sure what the correct answer is here but should the header be included in the `height`?

image

**Comment by maaslalani** [comment]
I guess it doesn't for `gum write` so perhaps this is good to be consistent and we can change it afterwards.

image

File: choose/options.go [hunk 1]
```diff
@@ -11,11 +11,13 @@ type Options struct {
...
 	Height            int          `help:"Height of the list" default:"10" env:"GUM_CHOOSE_HEIGHT"`
 	Cursor            string       `help:"Prefix to show on item that corresponds to the cursor position" default:"> " env:"GUM_CHOOSE_CURSOR"`
+	Header            string       `help:"Header value" default:"" env:"GUM_CHOOSE_HEADER"`
 	CursorPrefix      string       `help:"Prefix to show on the cursor item (hidden if limit is 1)" default:"○ " env:"GUM_CHOOSE_CURSOR_PREFIX"`
 	SelectedPrefix    string       `help:"Prefix to show on selected items (hidden if limit is 1)" default:"◉ " env:"GUM_CHOOSE_SELECTED_PREFIX"`
 	UnselectedPrefix  string       `help:"Prefix to show on unselected items (hidden if limit is 1)" default:"○ " env:"GUM_CHOOSE_UNSELECTED_PREFIX"`
 	Selected          []string     `help:"Options that should start as selected" default:"" env:"GUM_CHOOSE_SELECTED"`
 	CursorStyle       style.Styles `embed:"" prefix:"cursor." set:"defaultForeground=212" envprefix:"GUM_CHOOSE_CURSOR_"`
+	HeaderStyle       style.Styles `embed:"" prefix:"header." set:"defaultForeground=240" envprefix:"GUM_FILTER_HEADER_"`
```
Comment (maaslalani) [inline choose/options.go:20]:
```suggestion
	HeaderStyle       style.Styles `embed:"" prefix:"header." set:"defaultForeground=240" envprefix:"GUM_CHOOSE_HEADER_"`
```

**Review by maaslalani — APPROVED** [review]

**Comment by maaslalani** [comment]
As a heads up:

```
gum filter --header "Filter" --header.padding "2"
```

doesn't show the header correctly since we don't account for the header height when filter is auto height.

**Comment by MikaelFangel** [comment]
I'll give the padding issue a go later today or tomorrow if you don't beat me to it.

**Comment by maaslalani** [comment]
Awesome! No pressure / rush at all. The correct solution is to render the header with the style then calculate the height and subtract that so that you account for add possible height changes i.e. padding, margin, new lines, etc...

```
- lipgloss.Height(headerStyle.Render(header))
```

**Comment by MikaelFangel** [comment]
As you probably already knew. Rendering the header with the style before calculating the height resolved the issue.

Result with: `./gum filter --header "filter" --header.padding "2"`
result

**Comment by maaslalani** [comment]
Wonderful! Looks fantastic @MikaelFangel, thanks so much!

**Pull request merged**